File: app/src/main/java/com/jsoft/jcomic/helper/ComicsViewPager.java
package com.jsoft.jcomic.helper;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

import com.jsoft.jcomic.FullscreenActivity;

public class ComicsViewPager extends ViewPager {

    private FullscreenActivity activity;

    public ComicsViewPager(Context context) {
        super(context);
    }

    public void setActivity(FullscreenActivity activity) {
        this.activity = activity;
    }

    public ComicsViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void turnPrev() {
        if (this.getCurrentItem() == 0) {
            activity.episodeSwitch(-1);
        } else {
            activity.switchPageNum(this.getCurrentItem() - 1);
        }
    }

    public void turnNext() {
        int currentPageNum = this.getCurrentItem();
        this.setCurrentItem(this.getCurrentItem() + 1);

        if (currentPageNum == this.getCurrentItem()) {
            activity.episodeSwitch(1);
        } else {
            activity.switchPageNum(this.getCurrentItem());
        }
    }

    public void showPageBar() {
        activity.showPageBar();
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        // Never allow swiping to switch between pages
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        // Never allow swiping to switch between pages
        return false;
    }

    public void exitActivity() {
        activity.onBackPressed();
    }
}

File: app/src/main/java/com/jsoft/jcomic/helper/ComicsViewPager.kt
package com.jsoft.jcomic.helper

import android.content.Context
import androidx.viewpager.widget.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

import com.jsoft.jcomic.FullscreenActivity

class ComicsViewPager : ViewPager {

    private var activity: FullscreenActivity? = null

    constructor(context: Context) : super(context)

    fun setActivity(activity: FullscreenActivity) {
        this.activity = activity
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    fun turnPrev() {
        if (this.currentItem == 0) {
            activity!!.episodeSwitch(-1)
        } else {
            activity!!.switchPageNum(this.currentItem - 1)
        }
    }

    fun turnNext() {
        val currentPageNum = this.currentItem
        this.currentItem = this.currentItem + 1

        if (currentPageNum == this.currentItem) {
            activity!!.episodeSwitch(1)
        } else {
            activity!!.switchPageNum(this.currentItem)
        }
    }

    fun showPageBar() {
        activity!!.showPageBar()
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        // Never allow swiping to switch between pages
        return false
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        // Never allow swiping to switch between pages
        return false
    }

    fun exitActivity() {
        activity!!.onBackPressed()
    }

}